pkg/tls/mutal.go

```diff
@@ -0,0 +1,218 @@
+package tls
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"path/filepath"
+
+	"github.com/kris-nova/logger"
+
+	"github.com/spacemonkeygo/openssl"
+)
+
+const (
+	DefaultRSABytes int = 4096
+
+	//"/C=SP/ST=US/L=San Francisco/O=Default/OU=Default/CN=Root CA"
+)
+
+var (
+	DefaultX509SubjectFields = map[string]string{
+		"C":  "SP",
+		"ST": "US",
+		"L":  "San Francisco",
+		"O":  "Default",
+		"OU": "Default",
+		"CN": "Root CA",
+	}
+)
+
+type GRPCTLSGenerator struct {
+	RSABytes      int
+	Country       string
+	Organization  string
+	CommonName    string
+	SubjectFields map[string]string
+	CACert        *openssl.Certificate
+	ServerCert    *openssl.Certificate
+	ServerCSR     *openssl.Certificate
+	CAKey         openssl.PrivateKey
+	ServerKey     openssl.PrivateKey
+}
+
+func NewGRPCTLSGenerator(country, organization, name string) *GRPCTLSGenerator {
+	return &GRPCTLSGenerator{
+		RSABytes:      DefaultRSABytes,
+		Country:       country,
+		Organization:  organization,
+		CommonName:    name,
+		SubjectFields: DefaultX509SubjectFields,
+	}
+}
+
+func (g *GRPCTLSGenerator) Generate() error {
```

Comment:
This method I think needs the most work.

I was playing with how to take @leodido's work with the `openssl` CLI tool and replicate with this particular `openssl` library and was able to get this far.

The comments are off, and in general this could be cleaned up.